docs/user_guide/hdl_coding_guidelines.rst

```diff
@@ -0,0 +1,890 @@
+.. _hdl_coding_guidelines:
```

Comment:
I think we can link HDL's coding guidelines and have here only tests specific guidelines, e.g.
```
Tasks should have explicit port directions:
Wrong:

.. code::

   task my_task(int my_var);

Correct:

.. code::

   task my_task(input int my_var); 
```
Similar how the Docs Guidelines page works.
The rules themself we can organically add in future prs.